koala/ast/Range.py

```diff
@@ -0,0 +1,299 @@
+        else:
+            return Range(self.cells, map(lambda (key, value): value / other, enumerate(self.values())))
+
+    @staticmethod
+    def is_equal_one(self, other, ref):
+        first, second = get_values(ref, self, other)
+
+        return first == second
+
+    @staticmethod
+    def is_equal_all(self, other, ref = None):
+        if type(other) == Range:
+            return Range(self.cells, map(lambda (key, value): value == other.values()[key], enumerate(self.values())))
+        else:
+            return Range(self.cells, map(lambda (key, value): value == other, enumerate(self.values())))
+
+    @staticmethod
+    def is_not_equal_one(self, other, ref):
+        first, second = get_values(ref, self, other)
+
+        return first != second
+
+    @staticmethod
+    def is_not_equal_all(self, other, ref = None):
+        if type(other) == Range:
+            return Range(self.cells, map(lambda (key, value): value != other.values()[key], enumerate(self.values())))
+        else:
+            return Range(self.cells, map(lambda (key, value): value != other, enumerate(self.values())))
+
+    @staticmethod
+    def is_strictly_superior_one(self, other, ref):
+        first, second = get_values(ref, self, other)
+
+        return first > second
+
+    @staticmethod
+    def is_strictly_superior_all(self, other, ref = None):
+        if type(other) == Range:
+            return Range(self.cells, map(lambda (key, value): value > other.values()[key], enumerate(self.values())))
+        else:
+            return Range(self.cells, map(lambda (key, value): value > other, enumerate(self.values())))
+
+    @staticmethod
+    def is_strictly_inferior_one(self, other, ref):
+        first, second = get_values(ref, self, other)
+
+        return first < second
+
+    @staticmethod
+    def is_strictly_inferior_all(self, other, ref):
+        if type(other) == Range:
+            return Range(self.cells, map(lambda (key, value): value < other.values()[key], enumerate(self.values())))
+        else:
+            return Range(self.cells, map(lambda (key, value): value < other, enumerate(self.values())))
+
+    @staticmethod
+    def is_superior_or_equal_one(self, other, ref):
+        first, second = get_values(ref, self, other)
+
+        return first >= second
+
+    @staticmethod
+    def is_superior_or_equal_all(self, other, ref):
+        if type(other) == Range:
+            return Range(self.cells, map(lambda (key, value): value >= other.values()[key], enumerate(self.values())))
+        else:
+            return Range(self.cells, map(lambda (key, value): value >= other, enumerate(self.values())))
+
+    @staticmethod
+    def is_inferior_or_equal_one(self, other, ref):
+        first, second = get_values(ref, self, other)
+
+        return first <= second
+
+    @staticmethod
+    def is_inferior_or_equal_all(self, other, ref):
```

Comment:
Maybe we could factorize all these methods in 2 methods by passing a callback operator, but let's see it later